multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -0,0 +1,291 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import type { TestFn } from 'ava';
+import { AmountMath } from '@agoric/ertp';
+import type { Denom } from '@agoric/orchestration';
+import { divideBy } from '@agoric/zoe/src/contractSupport/ratio.js';
+import type { IBCChannelID } from '@agoric/vats';
+import { makeDoOffer, type WalletDriver } from '../../tools/e2e-tools.js';
+import { makeDenomTools } from '../../tools/asset-info.js';
+import { createWallet } from '../../tools/wallet.js';
+import { makeQueryClient } from '../../tools/query.js';
+import { commonSetup, type SetupContextWithWallets } from '../support.js';
+import { makeFeedPolicy, oracleMnemonics } from './config.js';
+import { makeRandomDigits } from '../../tools/random.js';
+
+const { keys, values, fromEntries } = Object;
+const { isGTE, isEmpty, make } = AmountMath;
+
+const test = anyTest as TestFn<
+  SetupContextWithWallets & {
+    lpUser: WalletDriver;
+    oracleWds: WalletDriver[];
+    nobleAgoricChannelId: IBCChannelID;
+    usdcOnOsmosis: Denom;
+  }
+>;
+
+const accounts = [...keys(oracleMnemonics), 'lp'];
+const contractName = 'fastUsdc';
+const contractBuilder =
+  '../packages/builders/scripts/fast-usdc/init-fast-usdc.js';
+
+test.before(async t => {
+  const { setupTestKeys, ...common } = await commonSetup(t);
+  const {
+    chainInfo,
+    commonBuilderOpts,
+    deleteTestKeys,
+    faucetTools,
+    provisionSmartWallet,
+    startContract,
+  } = common;
+  deleteTestKeys(accounts).catch();
+  const wallets = await setupTestKeys(accounts, values(oracleMnemonics));
+
+  // provision oracle wallets first so invitation deposits don't fail
+  const oracleWds = await Promise.all(
+    keys(oracleMnemonics).map(n =>
+      provisionSmartWallet(wallets[n], {
+        BLD: 100n,
```

Comment:
no IST? I would expect Oracle operators need IST.

I wonder when / where they get it...

Reply:
ah. right. They get an IST rebate when they pay for a smart wallet with BLD